docs/howto/types.md

```diff
@@ -0,0 +1,65 @@
+(types)=
+# How to prepare your Go types 
+
```

Comment:
Update the section titles so the structure of the HTG becomes clearer. Do you mean:

How to prepare your Go types for structs
How to prepare your Go types for maps
How to prepare your Go types for slices

?

PS "Prepare your Go types" is not a complete thought -- the story here is that you're preparing them for SQLair. That should be in all the titles. I'm also struggling to see where this prepare step fits in the overall story.